### PR TITLE
osd: fix suicide when osd bootup timeout

### PR DESCRIPTION
**Pull request by wyao51**
osd: suicide when osd bootup timeout
Fixes: https://tracker.ceph.com/issues/58182

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by wyao511** [comment]
lgtm

**Comment by tchaikov** [comment]
@wyao511 please more background of this change in the commit message, the title reads 

>  fix suicide when osd bootup timeout

i am afraid that this does not really help, not to mention it is more like a bug report.

**Comment by wyao511** [comment]
> @wyao511 please provide more background of this change in the commit message, the title reads
> 
> > fix suicide when osd bootup timeout
> 
> i am afraid that this does not really help, not to mention it is more like a bug report.



> @wyao511 please provide more background of this change in the commit message, the title reads
> 
> > fix suicide when osd bootup timeout
> 
> i am afraid that this does not really help, not to mention it is more like a bug report.

**Comment by wyao511** [comment]
> @wyao511 please provide more background of this change in the commit message, the title reads
> 
> > fix suicide when osd bootup timeout
> 
> i am afraid that this does not really help, not to mention it is more like a bug report.
>>
When the osd is started, if a message is lost, the OSD is stuck in the startup phase.
image

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -6147,6 +6158,29 @@ void OSD::tick_without_osd_lock()
 					      new C_Tick_WithoutOSDLock(this));
 }
 
+void OSD::bootup_timeout(int s, int t)
+{
+  dout(10) << "tick_timer bootup_timeout" << dendl;
+  int state = get_state();
+  if (s == state) {
+    dout(0) << "tick_timer bootup_timeout, osd has being " << get_state_name(state) << " state for " << t << " seconds, exit" << dendl;
```
Comment (ronen-fr) [inline src/osd/OSD.cc:6166]: derr, like the rest of the code that calls exit() in OSD.cc?

**Comment by ronen-fr** [comment]
LGTM (apart from the notes above)

**Pull request closed without merging**